packages/@aws-cdk/aws-lambda-nodejs/test/integ.dependencies.ts

```diff
@@ -24,6 +27,40 @@ class TestStack extends Stack {
   }
 }
 
+class SdkV3TestStack extends Stack {
+  public lambdaFunction: IFunction
+
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    // This function uses @aws-sdk/* but it will not be included
+    this.lambdaFunction = new lambda.NodejsFunction(this, 'external-sdk-v3', {
+      entry: path.join(__dirname, 'integ-handlers/dependencies-sdk-v3.ts'),
+      runtime: Runtime.NODEJS_18_X,
+    });
+  }
+}
+
 const app = new App();
-new TestStack(app, 'cdk-integ-lambda-nodejs-dependencies');
+const sdkV2testCase = new SdkV2TestStack(app, 'cdk-integ-lambda-nodejs-dependencies');
+const sdkV3testCase = new SdkV3TestStack(app, 'cdk-integ-lambda-nodejs-dependencies-for-sdk-v3');
+
+for (const testCase of [sdkV2testCase, sdkV3testCase]) {
+  const integ = new IntegTest(app, `LambdaDependencies-${testCase.stackId}`, {
+    testCases: [testCase],
+  });
+
+  const response = integ.assertions.awsApiCall('Lambda', 'invoke', {
+    FunctionName: testCase.lambdaFunction.functionName,
+  });
+  response.provider.addToRolePolicy({
+    Effect: 'Allow',
+    Action: ['lambda:InvokeFunction'],
+    Resource: [testCase.lambdaFunction.functionArn],
+  });
+  response.expect(ExpectedResult.objectLike({
+    StatusCode: 200,
+  }));
+}
```

Comment:
```suggestion
const integ = new IntegTest(app, 'LambdaDependencies', {
  testCases: [sdkV2testCase, sdkV3testCase],
});
for (const testCase of [sdkV2testCase, sdkV3testCase]) {
  const response = integ.assertions.awsApiCall('Lambda', 'invoke', {
    FunctionName: testCase.lambdaFunction.functionName,
  });
  response.provider.addToRolePolicy({
    Effect: 'Allow',
    Action: ['lambda:InvokeFunction'],
    Resource: [testCase.lambdaFunction.functionArn],
  });
  response.expect(ExpectedResult.objectLike({
    StatusCode: 200,
  }));
}
```

There should be one `IntegTest` per test, but it can have multiple test cases.

Reply:
@corymhall 
I tried your code, but not work. Because when we use `awsApiCall()` twice with same service and action, the ids of `AwsApiCall` construct is duplicated.

`AwsApiCall`'s id made by string literal as following:
```js
`AwsApiCall${service}${api}`
```
at [here](https://github.com/yamatatsu/aws-cdk/blob/2900339f5531b3ece2dd8c35af224a850f3fbe75/packages/@aws-cdk/integ-tests/lib/assertions/private/deploy-assert.ts#L62).

I'll create issue about it.

Reply:
Ah.. I realized it is work if we use `integ.assertions.invokeFunction()` or `new AwsApiCall()` instead of `integ.assertions.awsApiCall()`. Now I'm fixing.